### ISSUE TITLE
Angular-material and textAngular

### ISSUE THREAD

**Issue by vrghost242** [fraywing/textAngular#1097]
Desperatly trying to get an advanced editor working with Angular Material to allow for entering formatted text, thoguth textAngular might be JUST what I need. 
However, textAngular does not play well with angular-material. 

Load it as per instructions then opening the editor:
```` <md-input-container>
                            <div text-angular="text-angular" name="CV" ng-model="CV.CV" ta-disabled='enabled'></div>
                        </md-input-container>````

It does load, but it is broken, and looks special?

I am not certain if the below issue is the actual issue, but it is the only error I can find?


````Discontiguous selection is not supported.(anonymous function) @ textAngular-rangy.min.js:2
(anonymous function) @ textAngular-rangy.min.js:2
(anonymous function) @ textAngular-rangy.min.js:1p.init @ textAngular-rangy.min.js:1l @ textAngular-rangy.min.js:1
(anonymous function) @ textAngular.min.js:10invoke @ angular.js:4604
(anonymous function) @ angular.js:4412n @ angular.js:321fb @ angular.js:4412Ac.c @ angular.js:1691Ac @ angular.js:1712fe @ angular.js:1606
(anonymous function) @ angular.js:30423b @ angular.js:3108Rf @ angular.js:3398Qf.d @ angular.js:3386````

**Comment by vrghost242** [comment]
There was actually a second issue, as soon as I click any part of the editor I get
````
Uncaught DOMException: INDEX_SIZE_ERRda.getRangeAt 
@ textAngular-rangy.min.js:2i.getSelection 
@ textAngular.min.js:9v.on.l.events.mouseup 
@ textAngular.min.js:9Rf @ angular.js:3398Qf.d 
@ angular.js:3386o 
@ angular-material.min.js:7(anonymous function) 
@ angular-material.min.js:7r.end 
@ angular-material.min.js:7i 
@ angular-material.min.js:7f 
@ angular-material.min.js:7Rf 
@ angular.js:3398Qf.d 
@ angular.js:3386
````

**Comment by MikhailRoot** [comment]
Hi Guys! I've used TextAngular with AngularMaterial and all is ok in my case. But i didn't insert it inside md-input i've done it like this: 
`<md-card-content text-angular ng-model="vm.myRichText"></md-card-content>`

**Comment by ferc** [comment]
@MikhailRoot do you have an example? I have the same issue and I can't solve it (at least the versions of the libraries that you are using)

My example with the error: [JSFiddle](https://jsfiddle.net/fw81jgcx/)

**Comment by ferc** [comment]
Well, I found a temporary solution.

If you add jQuery before the AngularJS script, jQuery will manage the events and the bug is fixed.

This is a temporary solution because it adds the jQuery dependency and I don't want it.

I need a real solution for this, any ideas?

**Comment by pbassut** [comment]
@ferc What issue is that, exactly? I don't think @MikhailRoot raised an issue. Did he?

**Comment by ferc** [comment]
@pbassut if you run the [JSFiddle](https://jsfiddle.net/fw81jgcx/) in mobile or if you toggle the device mode in Chrome for example, when you click the textarea you will get this error:

*Uncaught DOMException: INDEX_SIZE_ERR*

So, text-angular is not compatible with Angular Material.

**Comment by pbassut** [comment]
Maybe you could look into [this pr](https://github.com/fraywing/textAngular/pull/1119) and see if it solves your issue?

**Comment by ferc** [comment]
No, I tested the changes and the PR doesn't solve this issue (maybe the PR solves other issues, but, not this one).

**Comment by ferc** [comment]
Line 1845 `var _selection = taSelection.getSelection();`
```javascript
element.on('mouseup', scope.events.mouseup = function(){
	var _selection = taSelection.getSelection();
	if(_selection.start.element === element[0] && element.children().length) taSelection.setSelectionToElementStart(element.children()[0]);
});
```

Line 691 `var range = rangy.getSelection().getRangeAt(0);`
```javascript
getSelection: function(){
	var range = rangy.getSelection().getRangeAt(0);
	var container = range.commonAncestorContainer;
```

**Comment by pbassut** [comment]
Note that after you take his PR, you should compile and only then link it to your html file.
Did you do this?

**Comment by ferc** [comment]
@pbassut I didn't need compile the files because to test I am using `bower_components/textAngular/dist/textAngularSetup.js` and `bower_components/textAngular/dist/textAngular.js` files

**Comment by pbassut** [comment]
Okay. I'll get a material design example working and post it here. 
Thanks for the information.

**Comment by MikhailRoot** [comment]
https://jsfiddle.net/mikhailroot/noLasa4v/3/ working example - but what i've mentioned really was tested on pc. And Yes when i've tested it inside Chrome in mobile view it throws an exception.

**Comment by ferc** [comment]
I can reproduce this error in mobile too (Android 4.4 / Chrome)

I added jQuery before AngularJS and I have a temporary fix, but, I don't need the jQuery dependency for another thing so I hope that exists a real solution for this

**Comment by HGXX** [comment]
Got the same error. Any solution out there ?

**Comment by ricardofilho6** [comment]
i was having the same issue, so i add font-awesome and it worked
